fip-4.md

```diff
@@ -0,0 +1,103 @@
+|tpid|Yes|FIO Address of TPID, See FIO Address validation rules|FIO Address of the wallet which generates this transaction. This FIO Address will be paid 10% of the fee.See FIO Protocol#TPIDs for details. Set to empty if not known.|
+|actor|Yes|FIO account name|FIO account for the signer, the account owning this payee FIO Address.|
+###### Example
+```
+{
+    "fio_address": "purse@alice",
+    "public_addresses": [{
+                          "chain_code": "BTC",
+                          "token_code": "BTC",
+                          "public_address": "1PMycacnJaSqwwJqjawXBErnLsZ7RkXUAs"
+                         },{
+                           "chain_code": "ETH",
+                           "token_code": "ETH",
+                           "public_address": "0xab5801a7d398351b8be11c439e05c5b3259aec9b"
+                         }
+                        ]
+    "max_fee": 0,
+    "tpid": "rewards@wallet",
+    "actor": "aftyershcu22"
+}
+```
+##### Processing
+* Verify the fio address format.
+* Verify the fio address exists. 
+* Verify the number of public addresses specified.
+* Verify the public addresses specified exist in the mappings on the chain
+* Verify that all information (chain code, token code and public address match the on chain mapping), error if failure
+* if any one of the public addresses does not exist on chain then error.
+* Check that fio_address is owned by actor.
+* require auth of the actor
+* verify that the fee for this does not exceed the max fee specified.
+* charge appropriate fee (this will be a bundled fee transaction, fee will be same as reject)
+* verify tx does not exceed max transaction size.
+* since this reduces state NO RAM BUMP.
+* remove all specified public addresses.
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid FIO address|Invalid format, or fio_address not found|400|"fio_address"|fio address invalid or not found|"Invalid FIO Address"|
+|Public addresses invalid|Public addresses contains information that does not exist, or does not match the on chain mappings|400|"public_addresses"|Invalid Public Addresses |
+|Fee exceeds maximum|fee exceeded the specified max|400|"max_fee"|fee exceeds max_fee specified|"Fee exceeds maximum"|
+|Invalid TPID|TPID is not empty or contains invalid FIO address|400|"tpid"|Value sent in is not empty and not a valid FIO Address format|"TPID must be empty or valid FIO address"|
+|Invalid Actor|Actor does not match payee FIO Address owner of request|403|||"Invalid Actor"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||fee_collected|String|fee amount collected SUFs|
+###### Example
+```
+{
+	
+			"status": "Ok",
+			"fee_collected": "0"		
+}
+```
+## Fees
+a new fee will be created remove_pub_address, type 1, 600000000 SUF
+
```

Comment:
Is it bundle eligible? If so how many bundled transactions does it use?

Reply:
I propose it use 1 bundled transaction. language will be added to the FIP to this effect.